### PR TITLE
made some major ish chnages

### PR DESCRIPTION
**Pull request by hsammon2**
added error handling for 404, 500, and, 405.  added a flash page so that dynamic messages based off of app.py can be sent to the user

added the profile path to the sql db

fixed logout function so that users can loggout of their account

adjusted the nave bar to reflect if a user is logged in or not.
    users not logged in cannot see logout or profile,

users logged in cannot se login or create account.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -31,6 +33,8 @@
 load_dotenv()
 app = Flask(__name__)
 
+
+#error handaler
```
Comment (tylerminnis) [inline app.py:37]: nice spelling

File: app.py [hunk 2]
```diff
@@ -11,13 +11,15 @@
 # ITSC 3155 - Software Engineering with Jacob Krevat 
 
 from dotenv import load_dotenv
-from flask import Flask, render_template, request, redirect, session
+from flask import Flask, render_template, request, redirect, session,  url_for, flash, abort
```
Comment (tylerminnis) [inline app.py:14]: yeah why flash?

File: .gitignore [hunk 3]
```diff
@@ -3,4 +3,7 @@ venv
 .DS_Store
 __pycache__
 .pytest_cache
-.env
\ No newline at end of file
+.env
+static/profile-pics/1-506C9D93-143C-4AA3-AF20-FE3A1C52C63D.JPG
+error_handlers.py
+package.json
```
Comment (tylerminnis) [inline .gitignore:9]: include node_modules por favor

**Review by tylerminnis — APPROVED** [review]
i think u should add some stuff like node_modules and the profile picture folder into gitignore but idc that much

other than that lgtm and ty for ur hard work

**Review by Nathan-Laney — APPROVED** [review]
Implemented Tyler's comment about gitignore changes. 
LGTM (lets get this merge!)

**Pull request merged**